framework/db/BaseActiveRecord.php

```diff
@@ -762,12 +780,11 @@ public function updateCounters($counters)
     {
         if ($this->updateAllCounters($counters, $this->getOldPrimaryKey(true)) > 0) {
             foreach ($counters as $name => $value) {
-                if (!isset($this->_attributes[$name])) {
-                    $this->_attributes[$name] = $value;
-                } else {
-                    $this->_attributes[$name] += $value;
+                if (null !== $current = $this->getAttribute($name)) {
```

Comment:
This is subjective, so if you want me to change it I will.

My reasoning:

($a = $b) == $c 

and

$a = $b == $c

have the same return value, but will result in diferent side effects. 
When reading either of them I have to think about brackets or operator priority. Whereas:

       $c == $a = $b 

Is unambiguous without brackets.